docs/memory.md

```diff
@@ -265,52 +285,33 @@ _Example_
 
 ```
 --memory size=0
---memory-zone size=1G,host_numa_node=0
-```
-
-### `guest_numa_node`
-
-Node identifier of a node that must be created in the guest. This option gives
-the user a way to create NUMA nodes in the guest and associate them with memory
-zones.
-
-This option can be very useful and powerful when combined with `host_numa_node`
-as it allows for creating a VM with non uniform memory accesses, and let the
-guest know about it. It allows for exposing memory zones through different NUMA
-nodes, which can help the guest workload run more efficiently.
-
-Value is an unsigned integer of 32 bits.
-
-_Example_
-
-```
---memory size=0
---memory-zone size=1G,guest_numa_node=0
+--memory-zone id=mem0,size=1G,host_numa_node=0
 ```
 
 ## NUMA settings
 
-Along with the guest NUMA nodes created through the `--memory-zone` parameter,
 `NumaConfig` or what is known as `--numa` from the CLI perspective has been
-introduced to define additional settings related to each NUMA node.
+introduced to define a guest NUMA topology. It allows for a fine description
+about the CPUs and memory ranges associated with each NUMA node. Additionally
+it allows for specifying the distance between each NUMA node.
 
 ```rust
 struct NumaConfig {
     id: u32,
     cpus: Option<Vec<u8>>,
     distances: Option<Vec<NumaDistance>>,
+    memory_zones: Option<Vec<String>>,
 }
 ```
 
 ```
---numa <numa>	Settings related to a given NUMA node "id=<node_id>,cpus=<cpus_id>,distances=<list_of_distances_to_destination_nodes>"
+--numa <numa>	Settings related to a given NUMA node "id=<node_id>,cpus=<cpus_id>,distances=<list_of_distances_to_destination_nodes>,memory_zones=<list_of_memory_zones>"
 ```
 
 ### `id`
 
-Node identifier of a guest NUMA node. The node referred by this identifier has
-been created through the `guest_numa_node` option from the `--memory-zone`
-parameter.
+Node identifier of a guest NUMA node. This identifier must be unique, otherwise
+an error will be returned.
```

Comment:
I think this is a bit confusing as everywhere else `id` is a string. Can we change this to `guest_numa_id` ?

Reply:
I agree, I can create an additional patch to take care of this.

Reply:
@rbradford this has been included into this PR, PTAL!